### PR TITLE
Enable cost based optimization by default

### PR DESCRIPTION
**Pull request by rongrong**
Switch join-distribution-type default value to AUTOMATIC.
Switch optimizer.join-reordering-strategy default value to AUTOMATIC.

```
== RELEASE NOTES ==

General Changes
* Enable cost based optimization by default.
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java [hunk 1]
```diff
@@ -70,7 +68,7 @@
     private double memoryCostWeight = 10;
     private double networkCostWeight = 15;
     private boolean distributedIndexJoinsEnabled;
-    private JoinDistributionType joinDistributionType = PARTITIONED;
+    private JoinDistributionType joinDistributionType = JoinDistributionType.AUTOMATIC;
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java:71]: static import AUTOMATIC

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java [hunk 2]
```diff
@@ -85,7 +83,7 @@
     private SingleStreamSpillerChoice singleStreamSpillerChoice = SingleStreamSpillerChoice.LOCAL_FILE;
     private String spillerTempStorage = "local";
     private DataSize maxRevocableMemoryPerTask = new DataSize(500, MEGABYTE);
-    private JoinReorderingStrategy joinReorderingStrategy = ELIMINATE_CROSS_JOINS;
+    private JoinReorderingStrategy joinReorderingStrategy = JoinReorderingStrategy.AUTOMATIC;
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java:86]: AUTOMATIC

**Review by highker — COMMENTED** [review]
The test failure seems to be related

**Comment by rschlussel** [comment]
We were just talking about enabling CBO by default! Generally looks good aside from @highker's comments.  Haven't looked at all the test failures, but definitely TestFeaturesConfig, PrestoCliTests, and TestJdbcConnection need to be updated with the correct defaults.

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java (hunk 2 above)
Comment (rongrong) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java:86]: Cannot static import 2 AUTOMATIC so I figured it's better to keep both not static import for clarity.

**Comment by rongrong** [comment]
@rschlussel @highker tests are fixed now. please review thanks!

File: presto-main/src/test/java/com/facebook/presto/sql/planner/TestPredicatePushdownWithDynamicFilter.java [hunk 3]
```diff
@@ -47,14 +47,14 @@ public void testNonStraddlingJoinExpression()
...
                         join(
                                 INNER,
-                                ImmutableList.of(equiJoinClause("ORDERS_OK", "LINEITEM_OK")),
-                                anyTree(tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
+                                ImmutableList.of(equiJoinClause("LINEITEM_OK", "ORDERS_OK")),
                                 anyTree(
-                                        filter(
-                                                "cast('2' as varchar) = cast(LINEITEM_LINENUMBER as varchar)",
+                                        node(
+                                                FilterNode.class,
```
Comment (rschlussel) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/TestPredicatePushdownWithDynamicFilter.java:53]: why did this change from specifying the filter?

**Review by rschlussel — APPROVED** [review]

**Review by highker — APPROVED** [review]

File: presto-main/src/test/java/com/facebook/presto/sql/planner/TestPredicatePushdownWithDynamicFilter.java (hunk 3 above)
Comment (rongrong) [reply presto-main/src/test/java/com/facebook/presto/sql/planner/TestPredicatePushdownWithDynamicFilter.java:53]: Dynamic filter will add a new predicate. It's a lot of work to make the filter matcher to be able to handle it. Existing dynamic filter tests are also just checking for FilterNode.

**Pull request merged**